addons/website_sale_collect/static/src/js/payment_form.js

```diff
@@ -0,0 +1,41 @@
+import { _t } from '@web/core/l10n/translation';
+import PaymentForm from '@payment/js/payment_form';
+
+PaymentForm.include({
+    /**
+     * Keep payment button context to switch labels displayed on payment method change.
+     *
+     * @override
+     */
+    async start() {
+        await this._super(...arguments);
+        const buttons = document.querySelectorAll('button[name="o_payment_submit_button"]');
+        this.paymentButtonContext = {
+            buttonLabelMapping: [...buttons].map((btn) => [btn, btn.textContent]),
+            payOnSiteLabel: _t("Confirm"),
+        };
+    },
+
+    /**
+     * Set a custom payment button label when `pay_on_site` method gets selected.
+     * Restore original label if method changes away from `pay_on_site`.
+     *
+     * @override
+     */
+    _selectPaymentOption(ev) {
+        this._super.apply(this, arguments);
+        const pmCode = this._getPaymentMethodCode(ev.target);
+        const { buttonLabelMapping, payOnSiteLabel } = this.paymentButtonContext;
+        if (pmCode === 'pay_on_site') {
+            for (const [button, _] of buttonLabelMapping) {
+                button.textContent = payOnSiteLabel;
+            }
+        } else {
+            for (const [button, label] of buttonLabelMapping) {
+                if (button.textContent === payOnSiteLabel) {
+                    button.textContent = label;
+                }
+            }
+        }
```

Comment:
due to the changes above, this also can be simplified
```suggestion
        const pmCode = this._getPaymentMethodCode(ev.target);
        this.buttons.forEach(button => {
             if (pmCode === 'pay_on_site') {
                 button.textContent = _t("Confirm");
             } else if (button.textContent !== this.submitButtonDefaultLabel) {
                 button.textContent = this.submitButtonDefaultLabel;
             }
         });
```

Reply:
Checking the pm code every iteration isn't much of a simplification imo. It's a constant value, so a single check should suffice.